quickwit/quickwit-ingest/src/ingest_v2/mrecordlog_utils.rs

```diff
@@ -104,8 +104,21 @@ pub(super) fn check_enough_capacity(
     Ok(())
 }
 
+/// Determines whether the queue is drained. A queue is considered drained if it is empty or has
+/// one EOF record.
+pub(super) fn is_queue_drained(mrecordlog: &MultiRecordLog, queue_id: &QueueId) -> bool {
+    mrecordlog
```

Comment:
This reminded me to open
https://github.com/quickwit-oss/quickwit/issues/4135

(We should discuss that some day.)